test/e2e/sagemaker/helper/trainingjob/xgboost-train.sh

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env bash
+
+##############################################
+# Tests for AWS SageMaker Training
+##############################################
+
+set -u
+set -x
+
+THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+ROOT_DIR="$THIS_DIR/../../../../.."
+SCRIPTS_DIR="$ROOT_DIR/scripts"
+
+source "$SCRIPTS_DIR/lib/common.sh"
+source "$SCRIPTS_DIR/lib/k8s.sh"
+source "$SCRIPTS_DIR/lib/testutil.sh"
+source "$SCRIPTS_DIR/lib/aws/sagemaker.sh"
+
+# Create a random trainingjob name
+# Expects RANDOM_ID env var to be set
+setup_sagemaker_training_test_inputs() {
+  if [[ -z ${RANDOM_ID+x} || -z ${AWS_REGION+x} || -z ${SAGEMAKER_DATA_BUCKET+x} ]]; then
+    echo "[FAIL] Usage setup_sagemaker_training_test_inputs. Expects RANDOM_ID, SAGEMAKER_DATA_BUCKET and AWS_REGION to be set"
+    exit 1
+  fi
+
+  TRAINING_JOB_NAME="ack-trainingjob-${RANDOM_ID}"
+  # TrainingJobName must have length less than 64
+  TRAINING_JOB_NAME=`echo $TRAINING_JOB_NAME|cut -c1-64`
+  TRAINING_IMAGE_REGISTRY=`echo $(get_xgboost_registry $AWS_REGION)`
+  TRAINING_IMAGE_USWEST2="$TRAINING_IMAGE_REGISTRY.dkr.ecr.$AWS_REGION.amazonaws.com/xgboost:1"
```

Comment:
nothing for this particular PR, but generally I shy away from using script-scoped or global-scoped variables like this inside a function. Better to just set these variables outside of a function in the test that is actually setting this up :)